woocommerce/woocommerce-android
# Support Order Details deep link

Closes: #7323


### Description

This PR adds support for deep links for opening orders details.

### Testing instructions


Please perform tests on the release variant (`vanillaRelease`). It's important because you'll test domain verification as well.

1. Prepare a few links:
```
1 https://woocommerce.com/mobile/orders/details?blog_id=<blog_A_id>&order_id=<order_A_id>

2 https://woocommerce.com/mobile/orders/details?blog_id=<blog_A_id>&order_id=<order_B_id>

3 https://woocommerce.com/mobile/orders/details?blog_id=<blog_B>&order_id=<order_C_id>

4 https://woocommerce.com/mobile/orders/details?blog_id=<random_numbers>&order_id=<random_numbers>

Not valid /*5 https://woocommerce.com/mobile/orders/details?blog_id=<blog_A_id>&order_id=<random_numbers>*/

6 https://woocommerce.com/mobile/orders/details?order_id=<random_numbers>

7 https://woocommerce.com/mobile/orders/details?blog_id=<random_numbers>
```

1. Click on the 1st link. Assert you can see `order A` on `blog A`
2. Click on the 2nd link. Assert you can see `order B` on `blog A`
3. Click on the 3rd link. Assert you can see `order C` on `blog B`. Then repeat test number 2. Make sure the app changes blogs correctly.
4. Click on the 4th link. Assert that the app just opened on the stats screen
5. Click on the 5th link. Assert that the app just opened on the stats screen
6. Click on the 6th link. Assert that the app just opened on the stats screen
7. Click on the 7th link. Assert that the app just opened on the stats screen

### Images/gif


https://user-images.githubusercontent.com/5845095/193794365-494f9e5f-e540-400a-a761-ad91306568d5.mov
